libs/libc/string/lib_memcmp.c

```diff
@@ -30,7 +30,9 @@
  * Public Functions
  ****************************************************************************/
 
-#ifndef CONFIG_LIBC_ARCH_MEMCMP
+#if !defined(CONFIG_LIBC_ARCH_MEMCMP) || \
```

Comment:
can we add a macro in libc/libc.h to avoid the dupliated?

Reply:
I agree, we should have something like:
```c
#if (!defined(CONFIG_LIBC_ARCH_USER_ONLY) && defined(__KERNEL__)) || \
    (!defined(CONFIG_LIBC_ARCH_KERNEL_ONLY) && !defined(__KERNEL__)
#  define LIBC_SHOULD_BUILD_FUNCTION 1
#endif
```

Reply:
> I agree, we should have something like:
> 
> ```c
> #if (!defined(CONFIG_LIBC_ARCH_USER_ONLY) && defined(__KERNEL__)) || \
>     (!defined(CONFIG_LIBC_ARCH_KERNEL_ONLY) && !defined(__KERNEL__)
> #  define LIBC_SHOULD_BUILD_FUNCTION 1
> #endif
> ```

That is the result for building arch-specific libc. Do you think we'd need a macro for both the software version and the arch's? As a drawback, we'd need to include libc.h on both sources (software and arch), too.

Reply:
Done!